modules/post/linux/gather/enum_containers.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Linux Container Enumeration',
+      'Description' => %q{
+        This module attempts to enumerate containers running on the target machine.
+        Supports Docker, LXC and RKT.
+      },
+      'License' => MSF_LICENSE,
+      'Author' => ['Mat Rollings [stealthcopter]'],
+      'Platform' => ['linux'],
+      'SessionTypes' => ['shell', 'meterpreter']
+    ))
+  end
+
+  # Check if a container program is installed and usable by current user
+  def runnable(container_type)
+    case container_type
+    when 'docker'
+      result = cmd_exec('docker ps >/dev/null 2>&1 && echo true')
+    when 'lxc'
+      result = cmd_exec('lxc list >/dev/null 2>&1 && echo true')
+    when 'rkt'
+      result = cmd_exec('rkt list >/dev/null 2>&1 && echo true')
+    else
+      print_error("Invalid container type #{container_type}")
+      return false
+    end
+    result =~ /true$/
+  end
+
+  # Count the number of currently running containers
+  def count_containers(container_type)
+    case container_type
+    when 'docker'
+      result = cmd_exec('docker ps --format "{{.Names}}" 2>/dev/null | wc -l')
+    when 'lxc'
+      result = cmd_exec('lxc list -c n --format csv 2>/dev/null | wc -l')
+    when 'rkt'
+      result = cmd_exec('rkt list 2>/dev/null | tail -n +2  | wc -l')
+    else
+      print_error("Invalid container type '#{container_type}'")
+      return 0
+    end
+    result.to_i
+  end
+
+  # List the currently running containers
+  def list_containers(container_type)
+    case container_type
+    when 'docker'
+      result = cmd_exec('docker ps')
+    when 'lxc'
+      result = cmd_exec('lxc list')
+    when 'rkt'
+      result = cmd_exec('rkt list')
+    else
+      print_error("Invalid container type '#{container_type}'")
+      return false
+    end
+    result
+  end
+
+  # Run Method for when run command is issued
+  def run
+    platforms = %w[docker lxc rkt]
+    platforms_found = false
+
+    platforms.each do |platform|
+      if runnable(platform)
+        platforms_found = true
+        no_active = count_containers(platform)
+        print_good("#{platform}: #{no_active} Active Containers")
+        if noActive > 0
+          containers = list_containers(platform)
+          print("#{containers}\n")
+        end
```

Comment:
Suggestion, but you could improve this code a little bit to have it print out something like `No active containers detected` via `print_warning` or similar when no containers are actually detected. This will make it stand out more and I don't think doing `print_good` when we have no actual containers is a good idea, as `print_good` is normally used when something good happens, like say we detect containers running. Finding no containers running would be more `print_warning` to warn the user that we couldn't find anything.

Also your `print` command here should be transformed into a `print_status` call here instead.

Reply:
Good point, will update!

Reply:
@gwillcox-r7 Just looking at changing this now and realized there was a reason using print_good however it probably wasn't clear enough. When it says 0 Active Containers it means that the container tool was found and is runnable by the current user, so it's a positive result even though there are no containers. Maybe this could be clearer with some different phrasing or something?

Reply:
Heyo @stealthcopter I was just thinking something like "Docker found but no containers were found to be running on it" would be better to prevent potential confusion? Otherwise the output makes it look like containers were found on first glance.

Reply:
good shout, have updated the code